app/src/main/java/de/westnordost/streetcomplete/osm/address/HouseNumbersParser.kt

```diff
@@ -27,7 +31,20 @@ fun parseHouseNumbers(string: String): HouseNumbers? {
 
 private fun parseHouseNumberParts(string: String): StructuredHouseNumber? {
     val c = StringWithCursor(string)
-    val houseNumber = c.nextMatchesAndAdvance("\\p{N}{1,5}".toRegex())?.value?.toIntOrNull() ?: return null
+    val houseNumber = c.nextMatchesAndAdvance("\\p{N}{1,5}".toRegex())?.value?.toIntOrNull()
+    if (houseNumber == null) {
+        val letterAtStart = c.nextMatchesAndAdvance("\\p{L}{1,2}".toRegex())?.value ?: return null
+        val separatorWithNumber = c.nextMatchesAndAdvance("(\\s?[/-]?\\s?)(\\p{N}{1,5})".toRegex())
+        return if (separatorWithNumber != null && c.isAtEnd()) {
+            HouseNumberWithLetterAtStart(
+                separatorWithNumber.groupValues[2].toInt(),
+                separatorWithNumber.groupValues[1],
+                letterAtStart
+            )
+        } else {
+            null
+        }
+    }
```

Comment:
This seems to be unnecessarily complicated. Why not

letterAtStart = ...
if (letterAtStart != null) separator = ... ?: ""
houseNumber = ....

if (letterAtStart != null ) return HouseNumberWithLetterAtStart(...)